### ISSUE TITLE
Route Announcer for Nuxt js 📢

### ISSUE THREAD

**Issue by sanjaiyan-dev** [nuxt/framework#6850]
### Describe the feature

Adding Route Announcer to Nuxt js for better accessibility.
Inspired by https://nextjs.org/docs/accessibility#route-announcements.

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://v3.nuxtjs.org/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

**Comment by sanjaiyan-dev** [comment]
Related PR-: https://github.com/nuxt/framework/pull/6847

**Comment by danielroe** [comment]
Off the bat, I love this idea. I think we should be providing tools to make it as easy as possible for developers to add essential accessibility features to their apps. Note that there are some judgement calls on how this is best implemented.

Related:
- https://github.com/vue-a11y/vue-announcer
